app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -76,17 +78,20 @@ class SparkRestClient(sparkConf: SparkConf) {
     .property(ClientProperties.READ_TIMEOUT, READ_TIMEOUT).target(historyServerUri).path(API_V1_MOUNT_PATH)
 
 
-  def fetchData(appId: String, fetchLogs: Boolean = false, fetchFailedTasks: Boolean = true)(
+  def fetchData(appId: String, fetchLogs: Boolean = false, retriesRemaining: Int = Int.MaxValue, fetchFailedTasks: Boolean = true)(
     implicit ec: ExecutionContext
   ): Future[SparkRestDerivedData] = {
     val (applicationInfo, attemptTarget) = getApplicationMetaData(appId)
-    val jobData = getJobDatas(attemptTarget)
+    val REQUEST_TIMEOUT =
```

Comment:
Can u please explain the logic : Ideally if the number of retries left are less , then timeout should be more .

Reply:
Yes, below is the same logic as you mentioned above. 
If the( retriesRemaining > permitted(3 in this case)) then Default timeout(5 secs) else 10 secs